### PR TITLE
Improve intel 1 to implement 1

### LINKED ISSUE #40: There should be a way to reset the reducer
**Describe the feature or change you'd like**
Currently it is not possible to reset the reducer without resetting the whole intel. This should be changed. It should be possible to reinit the reducer separately.

### LINKED ISSUE #63: The converters should be adjusted to prevent errors
**Describe the feature or change you'd like**
Currently the converters can crash if some specific data is not available. This breaks the whole intel.append() process.

Note to myself: Fix this.

**Additional context**
Ref: https://github.com/snek-at/intel/blob/implement-1/src/utils/github/converter.ts
Ref: https://github.com/snek-at/intel/blob/implement-1/src/utils/gitlab/converter.ts

### LINKED ISSUE #64: When appending a source list, it should be filtered by platform name
**Describe the feature or change you'd like**
Currently when attaching a source list with mixed Github and Gitlab orders, the source objects are attached by order. Since the basic data of the profile should be taken from a github source object, these must also be appended first.

To achieve this following changes are required:
- Sort the source list by platform name in order ["github", "gitlab"]

### LINKED ISSUE #65: There should be a way to get all GitHub organizations when generating talks
**Describe the feature or change you'd like**
It is currently not possible to filter organization by platform name. Therefore, all organizations are selected for the process of generating talks, including those from Gitlab. This leads to an error because the talk util currently only works with GitHub organizations.

To achieve this following changes are required:
- 1. Add the platformName to the organizations in reducer
- 2. Adjust query statements for organizations
- 3. Adjust platform name filter in talk generating process

### LINKED ISSUE #67: Is this indentation da wae? The SQL query formatting should fit a standard.
Is this indentation da wae? Is there a better way to display this without making it look like fucking Mary Poppins floating through time and fucking space with her umbrella.

_Originally posted by @Aichnerc in https://github.com/snek-at/intel/pull/66_

Ref: http://www.dpriver.com/pp/sqlformat.htm

### PR DESCRIPTION
**Pull request by schettn**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently the reducer reset is not working properly. #40 
- Currently it is common that the converters produce errors. #63
- Currently the order of source objects in a source list is to be observed. #64
- Currently when organizations with different platform types are present, the talks generation won't work because it can only handle organizations of GitHub. #65
- Currently the SQL formatting does not fit a standard. #67

**What is the new behavior (if this is a feature change)?**
- Now the reducer reset is working properly. Closes #40 
- Now all error sources are eliminated. Closes #63
- Now the order of source objects in a source list does not have to be considered. Closes #64 
- Now when organizations with different platform types are present the talks generation will only process GitHub organizations. Closes #65
- Now the SQL formatting does fit a standard. Closes #67

**Other information**:
As written in the coding guidelines snek-tonic, no more points are set at JSDoc headers. Exceptions include cases in which a comment spans several lines.

Over time, the intel should be adapted to the new coding guide line. #55

- Ref: 🐍

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -253,6 +253,18 @@ export class Intel implements IIntel {
...
    */
   async appendList(sources: ISource[]) {
+    /* Sources which platform name is github should be before others */
+    sources = sources.sort((a, b) => {
+      const platformNameOrder = ["github", "gitlab"];
+
+      const aPlatformNameIndex = platformNameOrder.indexOf(a.platform.name);
+      const bPlatformNameIndex = platformNameOrder.indexOf(b.platform.name);
+
+      //if (aPlatformNameIndex === bPlatformNameIndex) return a.card - b.card;
```
Comment (pinterid) [inline src/index.ts:263]: Is this still needed?

File: src/index.ts [hunk 2]
```diff
@@ -267,10 +279,12 @@ export class Intel implements IIntel {
   async generateTalks(sources: ISource[], organizations: string[] = []) {
     /* Check if organizations are provided */
     if (organizations.length === 0) {
-      /* Get all organization from database */
-      organizations = Reducer.models.Organization.objects.all();
+      /* Get all github organizations from database */
+      organizations = Reducer.models.Organization.objects.filter({
+        platformName: "github",
+      });
```
Comment (pinterid) [inline src/index.ts:285]: Add newline.

File: src/index.ts [hunk 3]
```diff
@@ -267,10 +279,12 @@ export class Intel implements IIntel {
...
       /* Convert to a organization name list */
-      organizations.map((organization: any) => {
+      organizations = organizations.map((organization: any) => {
```
Comment (pinterid) [inline src/index.ts:287]: Move this out of the if.

Comment (schettn) [reply src/index.ts:287]: I see no reason for this. Mapping should only take place if no organization name list is provided.

File: src/index.ts [hunk 4]
```diff
@@ -326,17 +340,16 @@ export class Intel implements IIntel {
    * @returns A reduced object.
    * @description Get a reduced object which contains profile and statistic data.
    */
-  get() {
+  async get() {
```
Comment (pinterid) [inline src/index.ts:343]: Is this needed.

Comment (schettn) [reply src/index.ts:343]: Yes it is. Without it get won't return a promise.

File: src/reducer/database/osm/index.ts [hunk 5]
```diff
@@ -2,10 +2,27 @@
...
+//> Reconstructor
+// SOAssembler for SO objects
+import { SOAssembler } from "./reconstructor";
 //#endregion
 
+//#region > Functions
+/**
+ * Reset all database entries.
+ * @function
+ * @description Delete all entries from all database tables
```
Comment (pinterid) [inline src/reducer/database/osm/index.ts:14]: Change to: "Delete all entries from every database table"

File: src/utils/upload/anonfiles.ts [hunk 6]
```diff
@@ -49,7 +49,7 @@ async function getDownloadUrl(url: string): Promise<string | null> {
  * @function
  * @param {Blob} file A file to be uploaded
  * @returns {ITalk} A unconverted talk object
- * @description Upload a file to anonfile and return 
+ * @description Upload a file to anonfile and return
```
Comment (pinterid) [inline src/utils/upload/anonfiles.ts:52]: The sentence is missing smth. return ?

File: src/reducer/database/osm/reconstructor.ts [hunk 7]
```diff
@@ -2,6 +2,7 @@
 //> AlaSQL
 // Contains a framework for creating a database in the browser.
 var alasql = require("alasql");
+
```
Comment (Aichnerc) [inline src/reducer/database/osm/reconstructor.ts:5]: Is it normal to have a new line before `#endregion`?

File: src/reducer/database/osm/statements/calendar.ts [hunk 8]
```diff
@@ -46,22 +46,27 @@ const betweenDate = `
...
+      AND total IN (
+                    SELECT 
+                      MAX(total)
+                    FROM
+                      calendar
+                    WHERE
+                      date >= "${from}"
+                        AND
+                          date <= "${to}"
+                    )
```
Comment (Aichnerc) [inline src/reducer/database/osm/statements/calendar.ts:69]: Is this indentation da wae?

File: src/reducer/database/osm/models.ts [hunk 9]
```diff
@@ -1,4 +1,10 @@
 //#region > Imports
+//> Moment
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:2]: Is there a need for a headline and a //#INSTALL?

Comment (schettn) [reply src/reducer/database/osm/models.ts:2]: Yea, anchors are defined in https://github.com/snek-at/tonic, but this anchor right there does not fit the standard. I will adjust this. One sec.

Comment (schettn) [reply src/reducer/database/osm/models.ts:2]: BTW. All other anchors will be set in the upcoming documentation update.

**Pull request merged**